saturdays-public-review/towers_of_hanoi.rb

```diff
@@ -33,33 +28,27 @@ def rules
   
   def offer_rules
     puts "Want to see the rules? (yes/no)"
-    positive = false
-    ans = gets
-    (1...ans.size).map do |i|
-      if ans[i] === 'y'
-        positive = true
-      end
-    end
-    if positive
+    answer = gets.chomp
+    if answer.start_with?('y')
       rules
     end
   end
   
   def sets
-    @sets = towers.map do |column| 
-      if column.empty? 
-        column = [" ", " ", " "]
-      elsif column.length < 3 && column.length > 1
-        column << " "
-      elsif column.length < 2 
-        column << " "
-        column << " "
-      else
-        column
+    sets = towers.map do |tower|
+      result = []
+      column = Array.new(@tower_size, " ")
+      column.each_index do |i|
+        if tower[i]
+          result[i] = tower[i]
+        else
+          result[i] = column[i]
+        end
       end
+      result
     end
-    @towers = @towers.map{|column| column.reject{|x| x == " "} }
-    @sets
+    self.towers = towers.map{|column| column.reject{|x| x == " "} }
```

Comment:
In case of a real project with larger classes and significant amount of variables and methods  _self.towers = towers_ might look confusing. In future try to find more clear way of assignment )